Scroll to bottom button Implemented

Prateek93a opened the pull request:
<!-- INSTRUCTION: Keep the line below to notify all core developers about this new PR -->
@RocketChat/ReactNative

<!-- INSTRUCTION: Inform the issue number that this PR closes, or remove the line below -->
Closes #1619

<!-- INSTRUCTION: Tell us more about your PR with screen shots if you can -->
I have implemented initial version of the button. It works fine as it should. I have used dimmer colors to avoid getting distracted by button while reading message. 
Right now I am using fixed offset to show the button, but if needed, we might use height of first message from bottom as in case of other apps.
If the feature seems useful, maybe we could implement it on ThreadView and SearchViews as well.

![ezgif com-crop(18)](https://user-images.githubusercontent.com/44807945/73062869-6870f700-3ec3-11ea-9b18-bdad1801d6f1.gif)


![Screenshot_20200124-160734_ DEVELOP  RocketChatRN](https://user-images.githubusercontent.com/44807945/73063018-d1f10580-3ec3-11ea-9732-059cfaab8a96.jpg)

diegolmello commented:
@Prateek93a Nice job. Can you resolve conflicts?

Prateek93a commented:
@diegolmello I have resolved the conflicts

diegolmello commented on app/views/RoomView/index.js:
```diff
@@ -783,6 +817,16 @@ class RoomView extends React.Component {
+		const { scrollButtonTranslateY, scrollToBottom } = this;
+		const { theme } = this.props;
+		return (
+			<AnimatedTouch theme={theme} onPress={scrollToBottom} style={[styles.scrollButton, { backgroundColor: themes[theme].bannerBackground, transform: [{ translateY: scrollButtonTranslateY }] }]}>
```
Add `borderColor: themes[theme].borderColor`

diegolmello commented on app/views/RoomView/index.js:
```diff
@@ -783,6 +817,16 @@ class RoomView extends React.Component {
+		const { theme } = this.props;
+		return (
+			<AnimatedTouch theme={theme} onPress={scrollToBottom} style={[styles.scrollButton, { backgroundColor: themes[theme].bannerBackground, transform: [{ translateY: scrollButtonTranslateY }] }]}>
+				<CustomIcon name='arrow-down' size={30} style={{ color: themes[theme].auxiliaryTintColor }} />
```
Can you use `tintColor` instead?

diegolmello commented on app/views/RoomView/styles.js:
```diff
@@ -46,5 +46,16 @@ export default StyleSheet.create({
+		borderRadius: 25,
+		alignItems: 'center',
+		justifyContent: 'center',
+		zIndex: 0
```
Add `borderWidth: 1`

diegolmello commented on app/views/RoomView/styles.js:
```diff
@@ -46,5 +46,16 @@ export default StyleSheet.create({
+	},
+	scrollButton: {
+		position: 'absolute',
+		bottom: 70,
```
This position should be based on device.
Take a look how it is hidden behind the keyboard on iPhone 11 Pro:
<img src="https://user-images.githubusercontent.com/804994/74954144-6e35fb80-53e1-11ea-9acc-e3147d3ad2a0.png" width="350" />

diegolmello commented on app/views/RoomView/index.js:
```diff
@@ -783,6 +817,16 @@ class RoomView extends React.Component {
 		return message;
 	}
 
+	renderScrollButton = () => {
```
After the changes, it's going to look like this:
<img src="https://user-images.githubusercontent.com/804994/74954344-ba813b80-53e1-11ea-8b47-6585f20c8b4d.png" width="350" />
<img src="https://user-images.githubusercontent.com/804994/74954397-cff66580-53e1-11ea-91f6-e2882303df08.png" width="350" />
<img src="https://user-images.githubusercontent.com/804994/74954419-da186400-53e1-11ea-843c-a0e8aafe2def.png" width="350" />

diegolmello commented on app/views/RoomView/List.js:
```diff
@@ -251,6 +252,9 @@ class List extends React.Component {
 					onEndReachedThreshold={5}
 					maxToRenderPerBatch={5}
 					windowSize={10}
+					scrollEventThrottle={20}
```
I'm unsure about performance here.
`scrollEventThrottle={20}` means it's going to execute `toggleScrollToBottomButton` every 20ms.
We already have some performance issues on low-end devices.
This implementation is much better than https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/735, but we still have to think about it.

Prateek93a replied:
@diegolmello  I have changed the number to 8

Prateek93a commented:
Ok Sir, on it.

Prateek93a commented:
I added listContainer because while typing, as the textbox height grew, the scroll button would get hidden behind the texbox since the `bottom` was set with respect to the bottom of the screen and not the textbox. So I added these changes.
Also I reduced the height of the textbox because since I had used listContainer, so now when textbox was at max height, the scrollbutton would get hidden behind header.
before-
![20200221_234727](https://user-images.githubusercontent.com/44807945/75060427-d3364200-5504-11ea-9cdb-5887270dd93d.jpg)
after-
![20200221_234711](https://user-images.githubusercontent.com/44807945/75060434-d6c9c900-5504-11ea-80b9-cca30c5e9509.jpg)

Prateek93a commented on app/views/RoomView/index.js:
```diff
@@ -912,8 +956,11 @@ class RoomView extends React.Component {
 					room={room}
 					renderRow={this.renderItem}
 					loading={loading}
+					animated={this.beginAnimating}
```
I have moved the component and logic inside list component
